Sources/Preferences/SwiftUIComponents.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  SwiftUIComponents.swift
+//  Preferences
+//
+//  Created by Kacper on 09/01/2020.
+//
+
+import SwiftUI
+
+/**
+    Extension with custom alignment guide for section title labels.
+ */
+@available (macOS 10.15, *)
+extension HorizontalAlignment {
+    private enum PreferenceSectionLabelAlignment: AlignmentID {
+        static func defaultValue(in context: ViewDimensions) -> CGFloat {
+            return context[HorizontalAlignment.leading]
+        }
+    }
+    
+    static let preferenceSectionLabel = HorizontalAlignment(PreferenceSectionLabelAlignment.self)
+}
+
+/**
+    Function builder for Preferences component used in order to restrict types of child view to PreferenceSection.
+*/
+@available (macOS 10.15, *)
+@_functionBuilder
+public struct SectionBuilder {
```

Comment:
This name is too generic for being public.
```suggestion
public struct PreferenceSectionBuilder {
```